application/src/main/java/org/thingsboard/server/service/sync/vc/DefaultEntitiesVersionControlService.java

```diff
@@ -456,18 +458,20 @@ public ListenableFuture<EntityDataDiff> compareEntityDataToVersion(SecurityUser
         EntityId externalId = ((ExportableEntity<EntityId>) entity).getExternalId();
         if (externalId == null) externalId = entityId;
 
-        return transformAsync(gitServiceQueue.getEntity(user.getTenantId(), versionId, externalId),
+        return transform(gitServiceQueue.getEntity(user.getTenantId(), versionId, externalId),
                 otherVersion -> {
                     SimpleEntitiesExportCtx ctx = new SimpleEntitiesExportCtx(user, null, null, EntityExportSettings.builder()
                             .exportRelations(otherVersion.hasRelations())
                             .exportAttributes(otherVersion.hasAttributes())
                             .exportCredentials(otherVersion.hasCredentials())
                             .build());
-                    EntityExportData<?> currentVersion = exportImportService.exportEntity(ctx, entityId);
-                    return transform(gitServiceQueue.getContentsDiff(user.getTenantId(),
-                            JacksonUtil.toPrettyString(currentVersion.sort()),
-                            JacksonUtil.toPrettyString(otherVersion.sort())),
-                            rawDiff -> new EntityDataDiff(currentVersion, otherVersion, rawDiff), MoreExecutors.directExecutor());
+                    EntityExportData<?> currentVersion;
```

Comment:
How is this related to message chunking and what does this fix?

Reply:
the ```getContentsDiff``` is now never used but is still present in the repository. why?

Reply:
JacksonUtil.toPrettyString is missing.

Reply:
Raw diff is not used but causes issues for huge entity data, as this diff might not fit into queue messages. JacksonUtil.toPrettyString was used for getting diff and is not needed now. Will remove from `getContentsDiff` method.